compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -109,35 +112,41 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
   override def transformTypeApply(tree: TypeApply)(implicit ctx: Context): Tree =
     transformTypeOfTree(tree)
 
-  /** If method overrides a Java varargs method, add a varargs bridge.
-   *  Also transform trees inside method annotation
+  /** If method overrides a Java varargs method or is annotated with @varargs, add a varargs bridge.
+   *  Also transform trees inside method annotation.
    */
   override def transformDefDef(tree: DefDef)(implicit ctx: Context): Tree =
     ctx.atPhase(thisPhase) {
-      if (tree.symbol.info.isVarArgsMethod && overridesJava(tree.symbol))
-        addVarArgsBridge(tree)
+      val isOverride = overridesJava(tree.symbol)
+      val hasAnnotation = hasVargsAnnotation(tree.symbol)
+      if tree.symbol.info.isVarArgsMethod && (isOverride || hasAnnotation) then
+        // java varargs can be overriden by synthetic methods with an Array param,
+        // but non-overrides need the varargs bytecode flag and cannot be synthetic
+        // otherwise javac refuses to call them.
+        addVarArgsBridge(tree, if hasAnnotation then JavaVarargs else Artifact)
       else
         tree
     }
 
   /** Add a Java varargs bridge
-   *  @param  ddef  the original method definition which is assumed to override
-   *                a Java varargs method JM up to this phase.
-   *  @return  a thicket consisting of `ddef` and a varargs bridge method
-   *           which overrides the Java varargs method JM from this phase on
-   *           and forwards to `ddef`.
+   *  @param  ddef   the original method definition which is assumed to override
```

Comment:
That comment needs to be updated since we can in fact call this on methods which don't override a Java method, right?

Reply:
yup, I forgot to update this one